### ISSUE TITLE
depending on mimemagic, licensing problem with GPL

### ISSUE THREAD

**Issue by oakbow** [y9v/carrierwave-base64#84]
https://github.com/carrierwaveuploader/carrierwave/issues/2548

>The gem mimemagic was previously licensed under MIT, but all existing versions were yanked this morning and replaced with new versions licensed under GPL in order to resolve a licensing issue (minad/mimemagic#97). The Rails community is currently working to update or replace this dependency, so that Rails apps are not required to be licensed under the GPL by default (rails/rails#41750).

like Carrierwave, carrierwave-base64 has same problems with [get_file_extension](https://github.com/y9v/carrierwave-base64/blob/515248b35a5e4dbb6445f7d76fe3c96659b7eca3/lib/carrierwave/base64/base64_string_io.rb#L50) method.

with this problem, we can not build/deploy/release our rails applications

**Comment by anhntq6** [comment]
now, my project can't deploy. @y9v Can you this fix error

**Comment by oakbow** [comment]
without thinking about licensing, you can deploy your project after executing `bundle update mimemagic`.
all old versions are yanked because of the license issue, but mimemagic 0.3.6 is released under GPL.
of course you need to consider it is valid to install GPL gem into your project

@anhntq6

**Comment by y-okamoto-1113** [comment]
@y9v 
i also have the same problem.
mimemagic 0.3.6 and 0.3.7 were yanked.
Instead, 0.3.8 and 0.3.9 and 0.4.2 were released.
https://github.com/mimemagicrb/mimemagic/commit/f14e3bfbcfdb212fca8975ef43d7d01e4c61d886


when i install mimemagic (0.4.2), a warning message was displayed.
Can you hepl me fix this warning.
```
Your lockfile doesn't include a valid resolution.
You can fix this by regenerating your lockfile or trying to manually editing the bad locked gems to a version that satisfies all dependencies.
The unmet dependencies are:
* mimemagic (~> 0.3.2), depended upon carrierwave-base64-2.8.0, unsatisfied by mimemagic-0.4.2
```

**Comment by oakbow** [comment]
how about using mimemagic 0.3.9 @y-okamoto-1113

**Comment by oakbow** [comment]
new version of Rails(Marcel) have been released which is removed mimemagic dependency.
now marcel has compatible methods with mimemagic.
I guess it is the most easy way to replace mimemagic with marcel, like carrierwave https://github.com/carrierwaveuploader/carrierwave/pull/2551

**Comment by justisb** [comment]
We need CarrierWave to release the new version with the Marcel dependency, but then I think this gem could be updated to also use Marcel fairly easily.

**Comment by oakbow** [comment]
I made a pull request for removing mimemagic dependency.

**Comment by justisb** [comment]
Once CarrierWave releases their update, this gem should also bump the minimum required `carrierwave` dependency version to match the release containing Marcel. Hopefully the CarrierWave gem gets a new release soon.

**Comment by y9v** [comment]
@oakbow @justisb unfortunately we have to wait for carrierwave to release an updated version first.

**Comment by oakbow** [comment]
carrierwave has been updated to 2.2.1. it has been free from mimemagic @y9v

**Comment by justisb** [comment]
@oakbow Perhaps you could re-submit your PR that was reverted, but with the carrierwave dependency bumped to `>= 2.2.1`.

Then after a new release of carrierwave-base64, everyone will just need to update carrierwave, carrierwave-base64, and rails gems to be free of mimemagic.

**Comment by oakbow** [comment]
@y9v 
I made a new PR to be free from mimemagic completely. https://github.com/y9v/carrierwave-base64/pull/87
please review it, thank you.

**Comment by y9v** [comment]
@justisb @oakbow thank you for your contribution, very much appreciated. I released 2.9.0, which does not depend on mimemagic anymore
